pkg/pkg.mk

```diff
@@ -6,18 +6,24 @@ PKG_BUILDDIR?=$(BINDIRBASE)/pkg/$(BOARD)/$(PKG_NAME)
 
 .PHONY: git-download
 
+ifneq (,$(wildcard $(PKG_DIR)/patches))
+git-download: $(PKG_BUILDDIR)/.git-patched
+else
 git-download: $(PKG_BUILDDIR)/.git-downloaded
+endif
 
-GIT_APPLY_PATCHES:=if test -d "$(PKG_DIR)"/patches; then \
-	git -C "$(PKG_BUILDDIR)" am --ignore-whitespace "$(PKG_DIR)"/patches/*.patch; \
-	fi
+ifneq (,$(wildcard $(PKG_DIR)/patches))
+$(PKG_BUILDDIR)/.git-patched: $(PKG_BUILDDIR)/.git-downloaded $(PKG_DIR)/Makefile $(PKG_DIR)/patches/*.patch
+	@git -C $(PKG_BUILDDIR) checkout -f $(PKG_VERSION) && \
+		git -C $(PKG_BUILDDIR) am --ignore-whitespace "$(PKG_DIR)"/patches/*.patch; \
+		touch $@
+endif
 
-$(PKG_BUILDDIR)/.git-downloaded:
-	rm -Rf $(PKG_BUILDDIR)
-	mkdir -p $(PKG_BUILDDIR)
-	$(GITCACHE) clone "$(PKG_URL)" "$(PKG_VERSION)" "$(PKG_BUILDDIR)"
-	$(GIT_APPLY_PATCHES)
-	touch $@
```

Comment:
I could revert the patch, but regardless of the fact if it fixes anything I think the patched version is safer, since it aborts in case of an error.